core/src/main/java/com/google/errorprone/bugpatterns/StatementSwitchToExpressionSwitch.java

```diff
@@ -99,6 +99,18 @@ public Description matchSwitch(SwitchTree switchTree, VisitorState state) {
     return NO_MATCH;
   }
 
+  private static boolean isRuleSwitch(SwitchTree switchTree) {
+    return switchTree.getCases().stream().anyMatch(StatementSwitchToExpressionSwitch::isRuleCase);
+  }
+
+  private static boolean isRuleCase(CaseTree caseTree) {
+    try {
+      return "RULE".equals(CaseTree.class.getMethod("getCaseKind").invoke(caseTree).toString());
+    } catch (ReflectiveOperationException e) {
+      throw new LinkageError(e.getMessage(), e);
+    }
+  }
```

Comment:
Given the [documented](https://github.com/openjdk/jdk/blob/67086ebf80109a623f3c2ad24d4e1a65de43d986/src/jdk.compiler/share/classes/com/sun/source/tree/CaseTree.java#L80-L87) invariant on which we now rely, perhaps it's more accurate to test for `"STATEMENT"`. But it seems unlikely that a third `Kind` will be introduced any time soon, and this way we avoid a negation at the call site. But I'm happy to switch this around.